sourcegraph/deploy-sourcegraph
# Implement kustomization files to base

### Checklist



* [ ] [CHANGELOG.md](https://github.com/sourcegraph/sourcegraph/blob/main/CHANGELOG.md) updated
* [ ] [K8s Upgrade notes updated](https://github.com/sourcegraph/sourcegraph/blob/main/doc/admin/updates/kubernetes.md)
* [ ] Sister [deploy-sourcegraph-docker](https://github.com/sourcegraph/deploy-sourcegraph-docker) change:
* [ ] All images have a valid tag and SHA256 sum

### Test plan



Close https://github.com/sourcegraph/customer/issues/693 & https://github.com/sourcegraph/sourcegraph/issues/43831

The changes made in this PR is supposed to work with the new [sourcegraph/kustomize](https://github.com/sourcegraph/kustomize) repo. 

By adding `kustomization.yaml` files directly to the base layer, users will no longer be required to clone this reference repository directly onto their machine to use kustomize. This is because [kustomize supports remote build](https://sourcegraph.com/github.com/kubernetes-sigs/kustomize@master/-/blob/examples/remoteBuild.md), where user can just reference this repository in their kustomization file ([example](https://github.com/sourcegraph/kustomize/blob/75a76bee1731a02fdf331caac9d88e599c38d0bd/overlays/non-privileged/kustomization.yaml)) through its URL.

Eliminating the needs of cloning this repository also prevents users from making changes directly to the base files.

## Test

To test the changes made in this PR:
1. create a new folder anywhere on your local machine. Name the folder `overlays`
2. Create a new file called `kustomization.yaml` inside the `overlays` folder
3. Copy and paste the code below to the `kustomization.yaml` file
4. `cd` back to one level before the `overlays` folder and run `kustomize build overlays`
5. you should see a new set of kustomized yaml files in your terminal output

```yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
namespace: ns-sourcegraph
resources:
  - git::git@github.com:sourcegraph/deploy-sourcegraph/base?ref=bee/kustomize
components:
  - git::git@github.com:sourcegraph/kustomize/components/non-privileged
  - git::git@github.com:sourcegraph/kustomize/components/non-privileged-create-cluster
```
